src/rules/importSpacingRule.ts

```diff
@@ -0,0 +1,124 @@
+/**
+ * @license
+ * Copyright 2013 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as ts from "typescript";
+
+import * as Lint from "../index";
+
+const LINE_BREAK_REGEX = /\n|\r\n/;
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "import-spacing",
+        description: "Ensures proper spacing between import statement keywords",
+        optionsDescription: "",
+        options: {},
+        optionExamples: [],
+        type: "style",
+        typescriptOnly: false,
+    };
+
+    public static ADD_SPACE_AFTER_IMPORT = "Add space after import";
+    public static TOO_MANY_SPACES_AFTER_IMPORT = "Too many spaces after import";
+    public static ADD_SPACE_AFTER_STAR = "Add space after *";
+    public static TOO_MANY_SPACES_AFTER_STAR = "Too many spaces after *";
+    public static ADD_SPACE_AFTER_FROM = "Add space after from";
+    public static TOO_MANY_SPACES_AFTER_FROM = "Too many spaces after from";
+    public static ADD_SPACE_BEFORE_FROM = "Add space before from";
+    public static TOO_MANY_SPACES_BEFORE_FROM = "Too many spaces before from";
+    public static NO_LINE_BREAKS = "No line breaks are allowed in import declaration";
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        const comparisonWalker = new ImportStatementWalker(sourceFile, this.getOptions());
+        return this.applyWithWalker(comparisonWalker);
+    }
+}
+
+class ImportStatementWalker extends Lint.RuleWalker {
+    private static IMPORT_KEYWORD_LENGTH = 6; // "import".length;
```

Comment:
The perf difference is not measurable. See the jsperf run https://jsperf.com/str-length